lib/puppet/configurer.rb

```diff
@@ -66,13 +66,17 @@ def initialize(transaction_uuid = nil, job_id = nil)
   # Get the remote catalog, yo.  Returns nil if no catalog can be found.
   def retrieve_catalog(facts, query_options)
     query_options ||= {}
-    result = retrieve_catalog_from_cache(query_options) if Puppet[:use_cached_catalog]
-    if result
-      @cached_catalog_status = 'explicitly_requested'
 
-      Puppet.info _("Using cached catalog from environment '%{environment}'") % { environment: result.environment }
-    elsif (@pluginsync_failed && result = retrieve_catalog_from_cache(query_options))
-      @cached_catalog_status = 'on_pluginsync_failure'
+    if Puppet[:use_cached_catalog] || @pluginsync_failed
+      result = retrieve_catalog_from_cache(query_options)
+    end
+
+    if result
+      if Puppet[:use_cached_catalog]
+        @cached_catalog_status = 'explicitly_requested'
+      elsif @pluginsync_failed
+        @cached_catalog_status = 'on_pluginsync_failure'
```

Comment:
Could you add the `on_pluginsync_failure` status to [`api/schemas/report.json`](https://github.com/puppetlabs/puppet/blob/f638777d6ece022dac8f8c2f6313870d494fc4aa/api/schemas/report.json#L113-L120) Both the `description` and `enum` fields?

We generally update the `report_format` (aka report version) field in the [schema](https://github.com/puppetlabs/puppet/blob/f638777d6ece022dac8f8c2f6313870d494fc4aa/api/schemas/report.json#L130) and [`Puppet::Transaction::Report`](https://github.com/puppetlabs/puppet/blob/f638777d6ece022dac8f8c2f6313870d494fc4aa/lib/puppet/transaction/report.rb#L227) whenever we serialize new fields, but I'm not 100% sure about adding a new value to an existing enumeration.

Reply:
Also would be good to add a test to `configurer_spec.rb` that the `on_pluginsync_failure` status is set when pluginsync fails.

Reply:
I added such test together with the implementation update: 
https://github.com/puppetlabs/puppet/blob/c1cd3f8c5fc121e2995c6acdcaa9252f0f827ca9/spec/unit/configurer_spec.rb#L107-L123